libqtile/widget/mpris2widget.py

```diff
@@ -97,16 +101,26 @@ def update(self, interface_name, changed_properties, invalidated_properties):
 
         metadata = changed_properties.get('Metadata')
         if metadata:
+            metadata = metadata.value
             self.is_playing = True
-            self.displaytext = ' - '.join([
-                metadata.get(x)
-                if isinstance(metadata.get(x), dbus.String)
-                else ' + '.join(y for y in metadata.get(x) if isinstance(y, dbus.String))
-                for x in self.display_metadata if metadata.get(x)
-            ])
+
+            meta_list = []
+            for key in self.display_metadata:
+                val = metadata.get(key)
```

Comment:
`val = getattr(metadata.get(key), 'value', None)` would save you 2 lines here but :shrug:

The only reason I even mention it is that these lines confused me on first read.

Reply:
Yes - not the easiest bit to read. Thanks for the suggestion. Will fix.